cmake/FindWayland.cmake

```diff
@@ -31,29 +31,41 @@ IF (NOT WIN32)
 
 	SET(WAYLAND_DEFINITIONS ${PKG_WAYLAND_CFLAGS})
 
-	FIND_PATH(WAYLAND_CLIENT_INCLUDE_DIR  NAMES wayland-client.h HINTS ${PKG_WAYLAND_INCLUDE_DIRS})
-	FIND_PATH(WAYLAND_SERVER_INCLUDE_DIR  NAMES wayland-server.h HINTS ${PKG_WAYLAND_INCLUDE_DIRS})
-	FIND_PATH(WAYLAND_EGL_INCLUDE_DIR     NAMES wayland-egl.h    HINTS ${PKG_WAYLAND_INCLUDE_DIRS})
+	FIND_PATH(WAYLAND_CLIENT_INCLUDE_DIR  NAMES wayland-client.h)
+	FIND_PATH(WAYLAND_SERVER_INCLUDE_DIR  NAMES wayland-server.h)
+	FIND_PATH(WAYLAND_EGL_INCLUDE_DIR     NAMES wayland-egl.h)
 	FIND_PATH(WAYLAND_CURSOR_INCLUDE_DIR  NAMES wayland-cursor.h HINTS ${PKG_WAYLAND_INCLUDE_DIRS})
 
-	FIND_LIBRARY(WAYLAND_CLIENT_LIBRARIES NAMES wayland-client   HINTS ${PKG_WAYLAND_LIBRARY_DIRS})
-	FIND_LIBRARY(WAYLAND_SERVER_LIBRARIES NAMES wayland-server   HINTS ${PKG_WAYLAND_LIBRARY_DIRS})
-	FIND_LIBRARY(WAYLAND_EGL_LIBRARIES    NAMES wayland-egl      HINTS ${PKG_WAYLAND_LIBRARY_DIRS})
-	FIND_LIBRARY(WAYLAND_CURSOR_LIBRARIES NAMES wayland-cursor   HINTS ${PKG_WAYLAND_LIBRARY_DIRS})
+	FIND_LIBRARY(WAYLAND_CLIENT_LIBRARIES NAMES wayland-client)
+	FIND_LIBRARY(WAYLAND_SERVER_LIBRARIES NAMES wayland-server)
+	FIND_LIBRARY(WAYLAND_EGL_LIBRARIES    NAMES wayland-egl)
+	FIND_LIBRARY(WAYLAND_CURSOR_LIBRARIES NAMES wayland-cursor)
 
-	set(WAYLAND_INCLUDE_DIR ${WAYLAND_CLIENT_INCLUDE_DIR} ${WAYLAND_SERVER_INCLUDE_DIR} ${WAYLAND_EGL_INCLUDE_DIR} ${WAYLAND_CURSOR_INCLUDE_DIR})
+	if (WAYLAND_CLIENT_INCLUDE_DIR AND WAYLAND_CLIENT_LIBRARIES AND NOT TARGET wayland::client)
+		add_library(wayland::client STATIC IMPORTED)
+		set_target_properties(wayland::client PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${WAYLAND_CLIENT_INCLUDE_DIR}" INTERFACE_LINK_LIBRARIES "${WAYLAND_CLIENT_LIBRARIES}")
+	endif()
+	if (WAYLAND_SERVER_INCLUDE_DIR AND WAYLAND_SERVER_LIBRARIES AND NOT TARGET wayland::server)
+		add_library(wayland::server STATIC IMPORTED)
+		set_target_properties(wayland::server PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${WAYLAND_SERVER_INCLUDE_DIR}" INTERFACE_LINK_LIBRARIES "${WAYLAND_SERVER_LIBRARIES}")
+	endif()
+	if (WAYLAND_EGL_INCLUDE_DIR AND WAYLAND_EGL_LIBRARIES AND NOT TARGET wayland::egl)
+		add_library(wayland::egl STATIC IMPORTED)
+		set_target_properties(wayland::client PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${WAYLAND_EGL_INCLUDE_DIR}" INTERFACE_LINK_LIBRARIES "${WAYLAND_EGL_LIBRARIES}")
+	endif()
+	if (WAYLAND_CURSOR_INCLUDE_DIR AND WAYLAND_CURSOR_LIBRARIES AND NOT TARGET wayland::cursor)
+		add_library(wayland::cursor STATIC IMPORTED)
+		set_target_properties(wayland::cursor PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${WAYLAND_CURSOR_INCLUDE_DIR}" INTERFACE_LINK_LIBRARIES "${WAYLAND_CURSOR_LIBRARIES}")
+	endif()
 
-	set(WAYLAND_LIBRARIES ${WAYLAND_CLIENT_LIBRARIES} ${WAYLAND_SERVER_LIBRARIES} ${WAYLAND_EGL_LIBRARIES} ${WAYLAND_CURSOR_LIBRARIES})
+	set(WAYLAND_LIBRARIES wayland::client wayland::server wayland::egl wayland::cursor)
```

Comment:
While this technically works, that's not what people expect to happen with CMake

Variables like `WAYLAND_LIBRARIES` are part of usage of old-style CMake, where people expect that to be a list of `.so` files, which they can link to while including everything in `WAYLAND_INCLUDE_DIRS`.

tbh we don't have to support old-style CMake usage, we should be using imported targets for our own code anyways